src/hotspot/share/sanitizers/address.hpp

```diff
@@ -76,6 +76,42 @@
   } while (false)
 #endif
 
+
+#ifdef ADDRESS_SANITIZER
+// This class uses RAII to temproraily unpoison a memory block.
+// Note that on dtor the memory block will be poisoned, regardless of its original un-/poisoned state.
+// General un/poisoning API are provided too.
+class AsanPoisoningHelper {
+  const void* _memory_region;
+  const size_t _size;
+ public:
+  AsanPoisoningHelper() = delete;
+  AsanPoisoningHelper(const void* addr, const size_t size) : _memory_region(addr), _size(size) {
+    ASAN_UNPOISON_MEMORY_REGION(_memory_region, _size);
+  }
+  ~AsanPoisoningHelper() {
+    ASAN_POISON_MEMORY_REGION(_memory_region, _size);
+  }
+  static void poison_memory(const void* addr, const size_t size) {
```

Comment:
Why are these functions needed, why not just call ASAN_POISON_MEMORY_REGION directly?